src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ImportController.php

```diff
@@ -236,6 +273,89 @@ public function downloadAction(Request $request)
         return $this->redirectToRoute('admin_import');
     }
 
+    /**
+     * Save import data match configuration.
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", message="You do not have permission to update this.", redirectRoute="admin_import")
+     * @DemoRestricted(redirectRoute="admin_import")
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function saveImportMatchAction(Request $request)
+    {
+        $formHandler = $this->get('prestashop.admin.import_data_configuration.form_handler');
+        $form = $formHandler->getForm();
+        $form->setData([
+            'match_name' => $request->request->get('match_name'),
+            'skip' => $request->request->get('skip'),
+            'type_value' => $request->request->get('type_value'),
+        ]);
+
+        $errors = $formHandler->save($form->getData());
+        $matches = [];
+
+        if (!$errors) {
+            $importMatchRepository = $this->get('prestashop.core.admin.import_match.repository');
+            $matches = $importMatchRepository->findAll();
+        }
+
+        return $this->json([
+            'errors' => $errors,
+            'matches' => $matches,
+        ]);
+    }
+
+    /**
+     * Delete import data match configuration.
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", message="You do not have permission to update this.", redirectRoute="admin_import")
+     * @DemoRestricted(redirectRoute="admin_import")
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function deleteImportMatchAction(Request $request)
+    {
+        $importMatchRepository = $this->get('prestashop.core.admin.import_match.repository');
+        $importMatchRepository->deleteById($request->get('import_match_id'));
+
+        return $this->json([]);
+    }
+
+    /**
+     * Get import data match configuration.
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getImportMatchAction(Request $request)
+    {
+        $importMatchRepository = $this->get('prestashop.core.admin.import_match.repository');
+        $importMatch = $importMatchRepository->findOneById($request->get('import_match_id'));
```

Comment:
What if `findOneById` finds nothing ?

Reply:
normally it cannot happen, because the user has to select an option from the list of available options. If somehow a non-existing option is selected, the repository would return false and nothing would be displayed to the user, which in my opinion is fine, since it was the same in legacy logic. We could improve the error handling in the future maybe, but I think that's not part of the migration process, wdyt?

Reply:
If we can clean and improve code during migration process, we should do it :) . A few cleaning and improvement is good (no big refactoring).

However here it's ok because the consequences of a non-existing option being selected are not bad (no php error, no JS error, no bad behavior).